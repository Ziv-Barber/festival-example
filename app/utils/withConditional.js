import React from 'react' // eslint-disable-line no-unused-vars

/**
 * High order component to display the component only if the condition is true.
 */
const withConditional = (Component) => ({
  condition,
  altComponent,
  ...props
}) => {
  if (condition) {
    return <Component {...props} />
  } // Endif.

  return altComponent || null
}

export default withConditional
