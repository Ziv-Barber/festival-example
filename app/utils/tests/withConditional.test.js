// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import withConditional from '../withConditional'

// For flow:
let { describe, it, afterEach, expect } = global

const TestComponent = () => <span>Example component</span>

/** @test {withConditional} */
describe('<withConditional />', () => {
  afterEach(cleanup)

  /** @test {withConditional} */
  it('should match snapshort', () => {
    const App = withConditional(TestComponent)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  /** @test {withConditional} */
  it('should render the component only when the condition passes', () => {
    const App = withConditional(TestComponent)
    const { getByText } = render(<App condition={true} />)
    expect(getByText(/^Example component/)).not.toBeNull()
  })

  /** @test {withConditional} */
  it('should return null when the condition fails', () => {
    const App = withConditional(TestComponent)
    const { queryByText } = render(<App condition={false} />)
    expect(queryByText(/^Example component/)).toBeNull()
  })
})
