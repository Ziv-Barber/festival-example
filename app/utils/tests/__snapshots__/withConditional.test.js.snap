// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withConditional /> should match snapshort 1`] = `<div />`;
